- url: 'https://github.com/clearpathrobotics/robot_upstart/pull/67'
  title: Add a dependency onto network-online.target
  body: "I Believe there should be a depend on networking.  I think in all cases networking should be up before ROS. I added a `After=` and `Wants=` with networking-online.target to the systemd template. This has been tested on a few of our robots and it seems to be working well. While avoiding any awkward timer delays which I've seen in the past.  \r\n\r\nDownside to waiting on network-online.target is possibly delaying the service if networking is misconfigured (even if this is true there is a 90 second timeout).   Waiting for network.taget may be safer.\r\n\r\n[NetworkTarget info](https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget)\r\n\r\nmy systemd critical path:\r\n\r\n```\r\nros.service @15.748s\r\n`-network-online.target @15.747s\r\n  `-network.target @15.735s\r\n    `-ifup@br0.service @584ms\r\n      `-apparmor.service @423ms +44ms\r\n        `-local-fs.target @418ms\r\n          `-boot-efi.mount @411ms +5ms\r\n            `-systemd-fsck@dev-disk-by\\x2duuid-EC37\\x2dF774.service @382ms +27ms\r\n              `-dev-disk-by\\x2duuid-EC37\\x2dF774.device @368ms\r\n```"
  repo: 'https://api.github.com/repos/clearpathrobotics/robot_upstart'
  labels: []
- url: 'https://github.com/clearpathrobotics/robot_upstart/issues/50'
  title: Support for Fedora
  body: "Right now, Fedora is disabled in the GBP tracks because there's no `daemontools` package on which it can depend. A few potential ways to eliminate the dependency:\r\n\r\n- Use the native capabilities of upstart/systemd to manage user switching—\_may cause issues with things like creating the log path, might have to move those to install time.\r\n- Use `su -s` instead of `setuidgid`, a la http://superuser.com/a/234541/48811\r\n- Bundle the tool, eg https://github.com/wertarbyte/coreutils/blob/master/src/setuidgid.c (note GPL)\r\n\r\nI'm not super motivated to work on this as I'm not a Fedora user so the status quo is acceptable to me. But the ticket is here as a marker of work to be done if someone wants to take it up."
  repo: 'https://api.github.com/repos/clearpathrobotics/robot_upstart'
  labels: []
- url: 'https://github.com/clearpathrobotics/robot_upstart/issues/45'
  title: >-
    ERROR: could not contact master [http://127.0.0.1:11311], when service is
    launched.
  body: "Hi, I'm new to robot_upstart package.\r\n\r\nFirst, my environment is below:\r\n\r\nubuntu mate 16.04 \r\nRaspberry Pi 3 \r\nros kinetic \r\n\r\nAnd I did that I installed robot upstart package from source at this repository (master branch).\r\nThen I install my launch file to run when  I RPI boot up.\r\n\r\n```\r\nrosrun robot_upstart install --job roscore foo/launch/bar.launch\r\n```\r\n\r\nBut I caught the folloing error. \r\n\r\n```\r\n1月 15 15:50:34 mal root[409]: roscore: Using workspace setup file /home/mal/catkin_ws/devel/setup.bash                                                      [676/1929]\r\n 1月 15 15:50:34 mal roscore-start[402]: <14>Jan 15 15:50:34 root: roscore: Using workspace setup file /home/mal/catkin_ws/devel/setup.bash\r\n 1月 15 15:50:37 mal root[558]: roscore: Launching ROS_HOSTNAME=mal, ROS_IP=, ROS_MASTER_URI=http://127.0.0.1:11311, ROS_LOG_DIR=/tmp\r\n 1月 15 15:50:37 mal roscore-start[402]: <14>Jan 15 15:50:37 root: roscore: Launching ROS_HOSTNAME=mal, ROS_IP=, ROS_MASTER_URI=http://127.0.0.1:11311, ROS_LOG_DIR=/tmp\r\n 1月 15 15:50:40 mal roscore-start[402]: ls: cannot access '/etc/ros/kinetic/roscore.d/*.xacro': No such file or directory\r\n 1月 15 15:50:41 mal root[639]: roscore: Generated launchfile: /tmp/roscore.launch\r\n 1月 15 15:50:41 mal roscore-start[402]: <14>Jan 15 15:50:41 root: roscore: Generated launchfile: /tmp/roscore.launch\r\n 1月 15 15:50:41 mal root[645]: roscore: Started roslaunch as background process, PID 644, ROS_LOG_DIR=/tmp\r\n 1月 15 15:50:41 mal roscore-start[402]: <14>Jan 15 15:50:41 root: roscore: Started roslaunch as background process, PID 644, ROS_LOG_DIR=/tmp\r\n 1月 15 15:50:44 mal roscore-start[402]: WARNING: cannot create log directory [/tmp/efe8ee0e-daee-11e6-a628-b827eb2a38c8]. Please set ROS_LOG_DIR to a writable location\r\n.\r\n 1月 15 15:50:44 mal roscore-start[402]: du: cannot read directory '/tmp/systemd-private-d841bace0f734c619ae55aa67b2e1288-systemd-hostnamed.service-FXeqHp': Permission \r\nd\r\n 1月 15 15:50:55 mal roscore-start[402]: No handlers could be found for logger \"roslaunch\"\r\n 1月 15 15:50:55 mal roscore-start[402]: ERROR: could not contact master [http://127.0.0.1:11311]\r\n 1月 15 15:50:55 mal roscore-start[402]: The traceback for the exception was written to the log file\r\n 1月 15 15:50:55 mal roscore-start[402]: Checking log directory for disk usage. This may take awhile.\r\n 1月 15 15:50:55 mal roscore-start[402]: Press Ctrl-C to interrupt\r\n 1月 15 15:50:55 mal roscore-start[402]: Done checking log file disk usage. Usage is <1GB.\r\n 1月 15 15:50:55 mal roscore-start[402]: [24B blob data]\r\n 1月 15 15:50:55 mal roscore-start[402]: started roslaunch server http://mal:40124/\r\n 1月 15 15:50:55 mal roscore-start[402]: SUMMARY\r\n 1月 15 15:50:55 mal roscore-start[402]: ========\r\n 1月 15 15:50:55 mal roscore-start[402]: PARAMETERS\r\n 1月 15 15:50:55 mal roscore-start[402]:  * /rosdistro: kinetic\r\n 1月 15 15:50:55 mal roscore-start[402]:  * /rosversion: 1.12.6\r\n 1月 15 15:50:55 mal roscore-start[402]: NODES\r\n 1月 15 15:50:55 mal roscore-start[402]:   /mal_upstart/\r\n 1月 15 15:50:55 mal roscore-start[402]:     integrator (mal_perception/integrator.py)\r\n 1月 15 15:50:55 mal roscore-start[402]: auto-starting new master\r\n 1月 15 15:50:55 mal roscore-start[402]: process[master]: started with pid [1015]\r\n 1月 15 15:50:55 mal roscore-start[402]: Traceback (most recent call last):\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/opt/ros/kinetic/bin/rosmaster\", line 35, in <module>\r\n 1月 15 15:50:55 mal roscore-start[402]:     rosmaster.rosmaster_main()\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/opt/ros/kinetic/lib/python2.7/dist-packages/rosmaster/main.py\", line 80, in rosmaster_main\r\n 1月 15 15:50:55 mal roscore-start[402]:     configure_logging()\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/opt/ros/kinetic/lib/python2.7/dist-packages/rosmaster/main.py\", line 58, in configure_logging\r\n 1月 15 15:50:55 mal roscore-start[402]:     _log_filename = rosgraph.roslogging.configure_logging('rosmaster', logging.DEBUG, filename=filename)\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/opt/ros/kinetic/lib/python2.7/dist-packages/rosgraph/roslogging.py\", line 126, in configure_logging\r\n 1月 15 15:50:55 mal roscore-start[402]:     logging.config.fileConfig(config_file, disable_existing_loggers=False)\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/logging/config.py\", line 85, in fileConfig\r\n 1月 15 15:50:55 mal roscore-start[402]:     handlers = _install_handlers(cp, formatters)\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/logging/config.py\", line 163, in _install_handlers\r\n 1月 15 15:50:55 mal roscore-start[402]:     h = klass(*args)\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/opt/ros/kinetic/lib/python2.7/dist-packages/rosgraph/roslogging.py\", line 167, in __init__\r\n 1月 15 15:50:55 mal roscore-start[402]:     from rospy.rostime import get_time, is_wallclock\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/__init__.py\", line 49, in <module>\r\n 1月 15 15:50:55 mal roscore-start[402]:     from .client import spin, myargv, init_node, \\\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/opt/ros/kinetic/lib/python2.7/dist-packages/rospy/client.py\", line 47, in <module>\r\n 1月 15 15:50:55 mal roscore-start[402]:     import yaml\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/dist-packages/yaml/__init__.py\", line 8, in <module>\r\n 1月 15 15:50:55 mal roscore-start[402]:     from loader import *\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/dist-packages/yaml/loader.py\", line 4, in <module>\r\n 1月 15 15:50:55 mal roscore-start[402]:     from reader import *\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/dist-packages/yaml/reader.py\", line 47, in <module>\r\n 1月 15 15:50:55 mal roscore-start[402]:     class Reader(object):\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/dist-packages/yaml/reader.py\", line 140, in Reader\r\n 1月 15 15:50:55 mal roscore-start[402]:     NON_PRINTABLE = re.compile(u'[^\\x09\\x0A\\x0D\\x20-\\x7E\\x85\\xA0-\\uD7FF\\uE000-\\uFFFD\\U00010000-\\U0010ffff]')\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/re.py\", line 194, in compile\r\n 1月 15 15:50:55 mal roscore-start[402]:     return _compile(pattern, flags)\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/re.py\", line 249, in _compile\r\n 1月 15 15:50:55 mal roscore-start[402]:     p = sre_compile.compile(pattern, flags)\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/sre_compile.py\", line 576, in compile\r\n 1月 15 15:50:55 mal roscore-start[402]:     code = _code(p, flags)\r\n 1月 15 15:50:55 mal roscore-start[402]:   File \"/usr/lib/python2.7/sre_compile.py\", line 561, in _code```\r\n```\r\n\r\nWhen I caught such error, I suspected timing of launching service.\r\nSo I add following line to the service file in /lib/systemd/system/roscore.service (in this case \"roscore\").\r\n\r\n```\r\n[Timer]\r\nOnBootSec=1min\r\n```\r\n\r\nAnd I reboot the PC, the service and launch file is launched correctly after a minute.\r\nI'm not sure why this happens.\r\n\r\nSorry, this is only report. but if I have some time to see the code. I will.\r\n\r\nThanks.\r\n\r\n\r\n"
  repo: 'https://api.github.com/repos/clearpathrobotics/robot_upstart'
  labels: []
- url: 'https://github.com/clearpathrobotics/robot_upstart/pull/44'
  title: Added "--env" option to add custom export statement in job-start
  body: "When \"--env\" option is specified, user can add custom export statement to job-start script using \"name=value\" format. This option supports setting multiple environment variables. If this option is not selected, no custom export will be added to job-start.\r\n\r\nExample usage\r\n```bash\r\nrosrun robot_upstart install ros_package/path/to/launch --env LIDAR_ENABLE=1\r\n```\r\nThis will add following line to just before launching launch file in job-start.\r\n```bash\r\nexport LIDAR_ENABLE=1\r\n```\r\n\r\nThis option is useful when launch file depends on specific environment variable to launch optional components.\r\n\r\nPlease review and provide feedback."
  repo: 'https://api.github.com/repos/clearpathrobotics/robot_upstart'
  labels: []
- url: 'https://github.com/clearpathrobotics/robot_upstart/issues/32'
  title: robot_upstart not finding install_script
  body: "I need to start ros launch file at machine startup .\r\n\r\nWe are using Ubilinux on Intel/Edison. We are using ROS:\r\n\r\n```\r\n/rosdistro: indigo\r\n/rosversion: 1.11.13\r\n```\r\n\r\nI am trying to follow these instructions http://docs.ros.org/api/robot_upstart... I have downloaded code from https://github.com/clearpathrobotics/... branch indigo-devel \r\nHere is how my folder looks\r\n\r\n```\r\nhemant@127:~/catkin_ws/src/robot_upstart/src/robot_upstart$ ls -ltr \r\ntotal 36\r\n-rw-rw-r-- 1 hemant hemant 2408 Dec 26 16:06 uninstall_script.py\r\n-rw-rw-r-- 1 hemant hemant 6667 Dec 26 16:06 providers.py\r\n-rw-rw-r-- 1 hemant hemant 9173 Dec 26 16:06 job.py\r\n-rw-rw-r-- 1 hemant hemant 4814 Dec 26 16:06 install_script.py\r\n-rw-rw-r-- 1 hemant hemant 1663 Dec 26 16:06 **init**.py\r\n```\r\n\r\nFollowing shows command I ran.\r\n\r\n```\r\nhemant@127:~/catkin_ws$ catkin_make\r\nhemant@127:~/catkin_ws$ catkin_make install\r\nhemant@127:~/catkin_ws$ source ~/catkin_ws/devel/setup.bash\r\n```\r\n\r\nNow creating a job:\r\n\r\n```\r\nhemant@127:~/catkin_ws$ rosrun robot_upstart install myrobot mybotpackage/test.launch\r\nTraceback (most recent call last):\r\n  File \"/home/hemant/catkin_ws/src/robot_upstart/scripts/install\", line 29, in <module>\r\n    from robot_upstart.install_script import main\r\nImportError: No module named robot_upstart.install_script\r\n```\r\n\r\nQuestion : Why it is not able to find robot_upstart.install_script\r\n\r\nI have been told to inform here  and  it could be an issue\r\nFor more information see here\r\n\r\nhttp://answers.ros.org/question/223129/robot_upstart-not-finding-install_script/\r\nThanks,\r\n"
  repo: 'https://api.github.com/repos/clearpathrobotics/robot_upstart'
  labels: []
- url: 'https://github.com/clearpathrobotics/robot_upstart/issues/27'
  title: Independent job for roscore
  body: >
    Would be nice when working remotely with a robot to be able to bounce the
    roslaunched stuff without getting a new roscore.
  repo: 'https://api.github.com/repos/clearpathrobotics/robot_upstart'
  labels: []
